kernel/src/hil/ble.rs

```diff
@@ -0,0 +1,20 @@
+use returncode::ReturnCode;
+
+// Defines the Interface between Capsules and Chips
+pub trait BleAdvertisementDriver {
+    fn start_adv(&self);
+    fn continue_adv(&self);
+    fn set_adv_data(&self,
+                    ad_type: usize,
+                    &'static mut [u8],
+                    len: usize,
+                    offset: usize)
+                    -> &'static mut [u8];
+    fn set_channel(&self, ch: usize);
+    fn set_adv_txpower(&self, dbm: usize) -> ReturnCode;
```

Comment:
It would be really good to comment each of these functions to explain their purpose. I'm happy to take some time and really think about this interface, but I'm not sure what your intent was for each of these functions and arguments.

See the [ADC HIL](https://github.com/helena-project/tock/blob/master/kernel/src/hil/adc.rs) for an example.